themes/default/content/docs/pulumi-cloud/esc/environments.md

```diff
@@ -235,245 +243,332 @@ values:
 
 ```yaml
 values:
   aws:
     login:
-      # provider definition ommited for brevity
+      fn::open::aws-login:
+        oidc:
+          roleArn: arn:aws:iam::01234567891011:role/some-role
+          sessionName: some-session
+
   environmentVariables:
-    AWS_ACCESS_KEY_ID: ${aws.creds.accessKeyId}
-    AWS_SECRET_ACCESS_KEY: ${aws.creds.secretAccessKey}
-    AWS_SESSION_TOKEN: ${aws.creds.sessionToken}
+    AWS_ACCESS_KEY_ID: ${aws.login.accessKeyId}
+    AWS_SECRET_ACCESS_KEY: ${aws.login.secretAccessKey}
+    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
     MY_ENV_VAR: "true"
 ```
 
-Now you can see those project environment variables if you open the environment with the `-f shell` or `-f dotenv` option.
+To render these values as environment variables in the shell, use `esc env open`, passing either `--format shell` or `--format dotenv`:
 
 ```bash
-$ esc open -f shell myorg/test
+$ esc env open myorg/test --format shell
 export AWS_ACCESS_KEY_ID="AKIAIOSFODNN7EXAMPLE"
-export AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
-export AWS_SESSION_TOKEN="eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE2OTY1NzA3NTIsImV4cCI6MTcyODEwNjc1MiwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSIsIkdpdmVuTmFtZSI6IkpvaG5ueSIsIlN1cm5hbWUiOiJSb2NrZXQiLCJFbWFpbCI6Impyb2NrZXRAZXhhbXBsZS5jb20iLCJSb2xlIjpbIk1hbmFnZXIiLCJQcm9qZWN0IEFkbWluaXN0cmF0b3IiXX0"
+export AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLE"
+export AWS_SESSION_TOKEN="eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOEXAMPLE"
 export MY_ENV_VAR="true"
 
-$ esc open -f dotenv myorg/test
+$ esc env open myorg/test --format dotenv
 AWS_ACCESS_KEY_ID="AKIAIOSFODNN7EXAMPLE"
-AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
-AWS_SESSION_TOKEN="eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE2OTY1NzA3NTIsImV4cCI6MTcyODEwNjc1MiwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSIsIkdpdmVuTmFtZSI6IkpvaG5ueSIsIlN1cm5hbWUiOiJSb2NrZXQiLCJFbWFpbCI6Impyb2NrZXRAZXhhbXBsZS5jb20iLCJSb2xlIjpbIk1hbmFnZXIiLCJQcm9qZWN0IEFkbWluaXN0cmF0b3IiXX0"
+AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLE"
+AWS_SESSION_TOKEN="eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOEXAMPLE"
 MY_ENV_VAR="true"
 ```
 
-You can evaluate those environment variables in your current shell with:
+To export rendered variables into the current shell, use `eval` or similar:
 
 ```yaml
-$ eval $(esc open -f shell myorg/test)
+$ eval $(esc env open myorg/test --format shell)
+$ echo $MY_ENV_VAR
+true
 ```
 
-## Running third party commands using Pulumi ESC secrets and config
+## Running commands with environment variables
+
+You can also run CLI commands directly, using environment variables obtained with Pulumi ESC --- without having to export those variables into the shell first.
```

Comment:
Hm. Well, considering this section is just about `esc run` and environment variables in general, I'm inclined to leave it as-is, since it does seem to work as described (both in this example and other commands). I've seen some discussion of oddness running `pulumi up` with Azure OIDC, though. Is that what you mean?

Reply:
Merging to get the rest of these updates out, but happy to address any follow-ups that come out this thread. (Thanks @toriancrane!)